src/Microsoft.Tye.Hosting/TyeDashboardApi.cs

```diff
@@ -262,6 +264,52 @@ private Task AllMetrics(HttpContext context)
         {
             var app = context.RequestServices.GetRequiredService<Application>();
 
+            var acceptHeader = context.Request.Headers[HeaderNames.Accept];
+
+            if (acceptHeader == MediaTypeNames.Application.Json)
+            {
+                context.Response.ContentType = MediaTypeNames.Application.Json;
+                var servicesMetricsCollection = CreateServicesMetricsCollectionJson(app);
+                return JsonSerializer.SerializeAsync(context.Response.Body, servicesMetricsCollection, _options);
+            }
+
+            context.Response.ContentType = MediaTypeNames.Text.Plain;
+            var response = CreateServicesMetricsCollectionText(app);
+            return context.Response.WriteAsync(response);
+        }
+
+        private Task Metrics(HttpContext context)
+        {
+            var app = context.RequestServices.GetRequiredService<Application>();
+
+            var name = (string?)context.Request.RouteValues["name"];
+
+            if (string.IsNullOrEmpty(name) || !app.Services.TryGetValue(name, out var service))
+            {
+                context.Response.StatusCode = 404;
+                return JsonSerializer.SerializeAsync(context.Response.Body, new
+                {
+                    message = $"Unknown service {name}"
+                },
+                _options);
+            }
+
+            var acceptHeader = context.Request.Headers[HeaderNames.Accept];
+
+            if (acceptHeader == MediaTypeNames.Application.Json)
```

Comment:
Requests can declare multiple acceptable response formats; should we support those cases (i.e. tease out whether the preferred response is JSON and, if so, use that, else fallback to text (assuming that's even in the acceptable list))?

Reply:
It seems to me that the Prometheus format for metrics is more expected. The json format may be needed in rare cases, for example, as I now find it more convenient to parse a json string than to work with plain text in my plugin. Most users, in my opinion, get metrics through specialized services that know how to work with the Prometheus format. So I think it's more correct to give json only if explicitly asked for it, and in all other cases, respond in Prometheus format.
What do you think?

Reply:
I'll buy that. :-)